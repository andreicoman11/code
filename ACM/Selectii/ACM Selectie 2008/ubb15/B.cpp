#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <sstream>
#include <map>
#include <list>
#include <cmath>
#include <algorithm>
#include <set>
#include <iomanip>
using namespace std;

ifstream fin ("B.in");
//ofstream fout ("B.out");

long double value(string);

int main()
{
	string e;
	getline(fin, e);
	while( e!="()" )
	{
		cout.setf(ios::fixed);
		cout << setprecision(2) << value(e) << "\n";
		getline(fin, e);
	}
	return 0;
}

long double value(string f)
{
	if( f[0]!='(' )
		return (long double)atoi(f.c_str());
	else
	{
		//cout << "f initial " << f << " ";
		f = f.substr(1, f.size()-2);
		//cout << "f fara paranteze " << f << "\n";

		istringstream in (f);
		double p;
		in >> p;

		f = f.substr( f.find(" ")+1 );
		//cout << "f fara p " << f << "\n";

		long double e1, e2;

		if( f[0]!='(' )
		{
			e1 = value( f.substr(0, f.find(" ") ) );
			e2 = value( f.substr( f.find(" ")+1 ) );
		}
		else
		{
			string f1;
			f1 += f[0];
			int p = 1, i;
			for(i=1; i<f.size() && p; i++)
			{
				f1 += f[i];
				if( f[i]=='(' ) p++;
				else if( f[i]==')' ) p--;
			}
			string f2 = f.substr(i+1);
			//cout << "f1 " << f1 << " --- f2 " << f2 << "\n";

			e1 = value(f1);
			e2 = value(f2);
		}

		//cout << e1 << " e1 --- e2 "  << e2 << "\n";
	
		long double rez;
		rez = p*(e1+e2) + (1-p)*(e1-e2);
		return rez;
	}
}