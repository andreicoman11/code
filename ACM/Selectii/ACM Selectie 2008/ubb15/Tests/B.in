0
1238
(0 100 1000)
(1 100 1000)
(0.066731 879468479 928858962)
(0.021070 -639909473 (0.204082 629457393 -500027120))
-798061733
(0.737414 (0.915231 562763941 -62143862) (0.612559 (0.498185 -984724380 424919545) 894659646))
226392505
-767495399
811009316
-673729864
289674419
-1014488064
(0.156725 141271893 (0.276388 60852928 273573283))
(0.820830 (0.360484 (0.690494 672739438 (0.688755 96166049 -588126851)) (0.178051 -873583400 785611771)) (0.576205 (0.885804 -870785285 -64785311) (0.909641 566381845 -595277471)))
199187385
-681658244
43805140
697316939
(0.702694 (0.487616 (0.463248 415938785 432975334) -318621457) (0.411881 536045494 -925597278))
(0.978938 (0.389728 (0.956953 -826703461 (0.388505 (0.010864 430814689 -506668905) (0.594304 577782568 (0.056388 124300197 -948984193)))) (0.854158 754789566 (0.466967 549477811 -1004691556))) (0.839178 756676402 606932131))
(0.440901 164877722 368224411)
(0.432277 -87893989 (0.911190 (0.896684 1045214656 594841984) (0.673985 362079225 (0.100658 (0.142679 747218121 -713998729) (0.858253 -829328403 (0.313384 227871997 -588838839))))))
()
