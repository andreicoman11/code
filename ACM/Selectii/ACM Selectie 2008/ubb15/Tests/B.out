0.00
1238.00
-900.00
1100.00
74576891.79
-1526304921.16
-798061733.00
138480968.80
226392505.00
-767495399.00
811009316.00
-673729864.00
289674419.00
-1014488064.00
183491703.99
426567062.26
199187385.00
-681658244.00
43805140.00
697316939.00
675440477.98
721625200.89
121354333.07
-398199747.88
